[#10, SEGAM-106] 로그인 페이지에서 상단바 제거

seunghui-park opened the pull request:
resolves #10

seunghui-park committed:
[#10] style: 로그인 페이지에서 탭바 제거

- 조건부 렌더링으로 로그인 페이지(/)에서는 탭바가 표시되지 않도록 변경
- usePathname 사용을 위해 루트 레이아웃을 클라이언트 컴포넌트로 변경

kmsu44 commented on src/app/(root)/layout.tsx:
루트레이아웃에서 해당 컴포넌트를 분리해도 좋지만, layout계층을 나누어 보면 좋을 것 같습니다

`app/layout`과
`app/(root)/layout`으로 분리하면 isLoginPage와 같은 conditional렌더링을 사용하지 않아도 될 것 같아요!

seunghui-park replied:
계층 분리도 생각 했었는데 그 방법이 더 좋군요 !
바로 수정하겠습니다 !!

kmsu44 replied:
이 방법이 더 좋은건 아니에요!! 개인의 취향이 존재할 뿐입니다...!!
제가 코멘트 다는 것들은 참고정도로만...

++
추가적으로 server 컴포넌트와 client컴포넌트의 렌더링 관계를 알아 두시면 좋습니다
https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#client-components

seunghui-park replied:
감사합니다! 😊

seunghui-park committed:
[#10] refactor: 레이아웃 계층 분리로 로그인 페이지에서 탭바 제거

- 로그인 페이지는 (auth) 레이아웃을 사용하고, 공지사항 및 스터디룸 페이지는 (root) 레이아웃을 사용하도록 변경
- global.css가 모든 레이아웃에 적용되도록 구조 조정

kmsu44 reviewed (commented):
html을 포함한 layout은 (auth)와 (root)공통으로 사용할 수 있도록 구성해보시면 좋을 듯 합니다

seunghui-park committed:
[#10] refactor: 공통 레이아웃 분리 및 로그인 페이지 구조 개선

- 최상위 레이아웃에서 html과 body 태그 관리
- (auth)와 (root)에서 공통 레이아웃 재사용하도록 변경

seunghui-park commented:
늦은 시간에 죄송합니다 🥲 피드백 반영해서 공통 레이아웃 적용했습니다!

kmsu44 commented:
> 늦은 시간에 죄송합니다 🥲 피드백 반영해서 공통 레이아웃 적용했습니다!

아닙니다!!!

다만... nextjs 폴더 위계 자체가 다음과 같은 구조를 가지고 있어서 <Layout/>처럼 import를 하지 않아도 사용이 가능합니다.
https://nextjs.org/docs/app/getting-started/project-structure#component-hierarchy

따라서 지금 작성하신 구조에서 AuthLayout은 추가로 작성하지 않아도 괜찮아요!!
해당 글 읽어 보시면 이해하시는데 도움이 될 거에요:)

tnqkr3494 commented on src/app/(root)/layout.tsx:
```diff
@@ -1,16 +1,11 @@
-}>) {
+export default function RootLayout({children}: { children: React.ReactNode }) {
   return (
-    <html lang="en">
```
마찬가지로 이 부분도 Layout을 사용하지 않고 fragment 처리해도 됩니다! 직관적으로 보기에는 Layout을 사용하는 것도 괜찮은것 같습니다.

tnqkr3494 commented on src/app/(root)/layout.tsx:
```diff
@@ -1,16 +1,11 @@
-      <body className="bg-gray-100">
-        <nav className="flex items-center justify-between px-8 py-2 shadow-md border-b border-gray-300 safe-area-top">
-          <h2 className="f16 font-bold text-theme_primary">Segam Admin</h2>
+    <Layout>
```
마찬가지로 이 부분도 Layout을 사용하지 않고 fragment처리해도 괜찮습니다! 직관적으로 보기에는 Layout도 괜찮아 보입니다.

tnqkr3494 commented:
또한 추후 어드민 기능이 추가될 경우 관리하기 편하기 위해서는, 상단바에 대한 코드를 따로 컴포넌트쪽으로 분리시켜도 괜찮은 방법인 것 같습니다.

seunghui-park commented:
> 또한 추후 어드민 기능이 추가될 경우 관리하기 편하기 위해서는, 상단바에 대한 코드를 따로 컴포넌트쪽으로 분리시켜도 괜찮은 방법인 것 같습니다.

넵 그게 좋겠네요 ! 컴포넌트로 분리하고 레이아웃에서 import하는 식으로 리팩토링 하겠습니다!

seunghui-park committed:
[#10] refactor: 어드민 상단바 컴포넌트 분리

tnqkr3494 commented:
고생하셨습니다!
